pkg/util/proxy_test.go

```diff
@@ -162,3 +162,19 @@ func Test_http_proxy_from_env_lowercase_precedence(t *testing.T) {
 	assert.Equal(t, "http://user:pass@someotherproxy.foo:1080", proxyConfig.httpProxy)
 	assert.Equal(t, "https://user:pass@someotherproxy.foo:1080", proxyConfig.httpsProxy)
 }
+
+func Test_parse_special_character_uri(t *testing.T) {
+	var urlList = []struct {
+		givenURI    string
+		expectedURI string
+	}{
+		{"", ""},
+		{"http://foo.com:3128", "http://foo.com:3128"},
+		{"http://user:F@oo!B#ar$@myserver:3128", "http://user:F@oo%21B%23ar$@myserver:3128"},
+		{"https://myuser:my#pass@foo.com:3128", "https://myuser:my%23pass@foo.com:3128"},
+	}
+	for _, proxyUrl := range urlList {
+		got := parseProxySpecialChar(proxyUrl.givenURI)
+		assert.Equal(t, proxyUrl.expectedURI, got)
+	}
+}
```

Comment:
👍

Reply:
To run the unit test on my local setup i am following the the upstream doc [1]. I am not sure what would be the ```<Testcase Name>``` for the command i wand to execute a single test on a single package
```
go test -v pkg/util/proxy_test.go -run <Testcase Name>
```
[1] https://docs.openshift.org/latest/minishift/contributing/developing.html#unit-tests

Reply:
@amitkrout here is the right way to run it. You need to point to the the package directory i.e. `util`. 

```
$ go test -v ./pkg/util -run Test_parse_special_character_uri
=== RUN   Test_parse_special_character_uri
--- PASS: Test_parse_special_character_uri (0.00s)
PASS
ok  	github.com/minishift/minishift/pkg/util	0.017s
```

Reply:
Passing for me as well.